src/Signature/SignatureTrait.php

```diff
@@ -0,0 +1,49 @@
+<?php
+namespace Aws\Signature;
+
+/**
+ * Provides signature calculation for SignatureV4.
+ */
+trait SignatureTrait
+{
+    /** @var array Cache of previously signed values */
+    private $cache = [];
+
+    /** @var int Size of the hash cache */
+    private $cacheSize = 0;
+
+    private function createScope($shortDate, $region, $service)
+    {
+        return "$shortDate/$region/$service/aws4_request";
+    }
+
+    public function getSigningKey($shortDate, $region, $service, $secretKey)
```

Comment:
I don't think this should be a public method. It was private in the `SignatureV4` class.